#102 Update FAQ page (#113)
Inbibo/adonisfx_docs: docs/faq.md

Similarly to a skin simulation setup, the fascia can be simulated using [AdnSkin](maya/skin) deformer as well. In this case, it is recommended to use values of *Rest Length Multiplier* lower than 1. This deformer also requires one mesh (a single target, i.e. the muscle geometries combined) or multiple meshes (multiple targets, i.e. the muscle geometries separated) to drive the simulation and a mesh to apply the deformer to (fascia geometry).

## How can I generate the fascia geometry?

In order to obtain a plausible fascia geometry, we can use two techniques that require the muscle geometries and the skin geometry as inputs.

> The easiest option is to use the shrinkwrap deformer provided by Maya. This solution works better for low poly geometries.

Reviewer: ```suggestion
The first approach is to use the shrinkwrap deformer provided by Maya. This solution should produce good results for low poly geometries.
```

Author: Done.

- Create a copy of the skin geometry that will become your fascia.

> - Select the combined geometry of all muscles, then select the copy of the skin geometry.

Reviewer: Split into 2 steps please:
- Combined all the muscles together
- Select the combined muscles geometry, then select the copy of the skin geometry.

Author: Done.

- Apply the shrinkwrap deformer and adjust its settings until the desired fascia geometry is achieved.
- Delete the history of the fascia geometry.

> For high-end fascia simulations, a better option would be to create the fascia geometry using a pressured cloth simulation. nCloth can be used to achieve this:

Reviewer: ```suggestion
Ths second approach is to use nCloth to apply a pressured simulation. This solution could work better for high-end fascia simulations. The procedure is the following:
```

Author: Done.

- Create a copy of the skin geometry that will become your fascia.
- Select the fascia geometry and click nCloth > Create nCloth.

> - Select the muscle geometries and click on nCloth > Create Passive Collider.

Reviewer: I think you should combine first, not sure. Can you please double check? If combining is needed, please add a step before.

Reviewer 2: Correct! Also, in my setups I also included the bones in some cases to avoid more intersections. I would add that.
And optionally I also tweaking the friction and stickiness parameters on the nCloth to avoid the cloth simulation to slide too much over the muscles and fold. I let you decide if we add that or not @carlosmonteagudoinbibo

Author: Added the step for combining the muscles. Let me know if we should add more details like including bones or changing other parameters in nCloth. @carlosmonteagudoinbibo

Reviewer: Good point again, let's double check with Francesco and let's follow his suggestion.

Reviewer 2: CC: @francescogioiainbibo

> - Select the nucleus node and set the gravity to 0.

Reviewer 2: ```suggestion
- Select the nucleus node and set the gravity to 0.0.
```

Author: Done.

- Select the nCloth node and keyframe the pressure attribute to 0 at the first frame of the scene.
- Advance some frames (e.g. ten frames) and keyframe the pressure to -1.
- Select the fascia geometry and click nCloth > Create New nCache > nObject.

> - Once the simulation finishes, choose the fascia that looks best within the simulated frame range.

Reviewer: I think we have to delete the history too in order to get a clean static geometry. Could you please double check?

Reviewer 2: Correct! I would also advise something for later. As we require the fat layer to have the same topology as the skin layer it is importand that the fascia is subdivided the same way as the fat layer (skin duplicated essentially). And maybe we can suggest to add some DeltaMush to smoothen even firther the fascia? Maybe that can be optional. CC: @carlosmonteagudoinbibo

Author: Added a step for deleting the history. Also added an optional step for increasing the subdivisions on the fascia if more resolution is needed. Let me know if we should add a step related to applying DeltaMush. @carlosmonteagudoinbibo.

Reviewer: I see the benefits of smoothing, but let's check with Francesco. If it is something CFX artist use to do, yes, add it. Otherwise let's not add it.

Reviewer 2: CC: @francescogioiainbibo

## How can I simulate fat?

Fat can be simulated using the [AdnFat](maya/fat) deformer. This deformer requires one base mesh (i.e. the fascia) to drive the fat simulation and a mesh to apply the deformer to (fat mesh). Both meshes must have the same vertex count and triangulation. A simple setup is explained [here](maya/simple_setup#adnfat).

## Can I simulate muscles, fascia, fat and skin all coupled?

Yes, you can simulate muscles, fascia, fat and skin following these steps:

- Configure every muscle geometry with an AdnMuscle deformer.
- Combine all muscle geometries into a single geometry.

> - Shrinkwrap the skin geometry to the combined muscles geometry to obtain the fascia geometry. Alternatively, pressure can be simulated on the skin geometry so that it is adjusted to the muscle geometries to obtain the fascia geometry.

Reviewer: I suggest:

```suggestion
- Generate the fascia geometry. Check this answer to know how to complete this step.
```

And make "this answer" a link to the question above.

Important!! The step "Combine all muscle geometries into a single geometry" has to be removed. This is not a requirement anymore from version v1.2.0 actually

Reviewer 2: It still is a requirement for the shrinkwrap though, not for setting up the fascia. We should differentiate. For the step below "Configure the fascia with an AdnSkin deformer: Select the combined muscles geometry," and some other steps we should remove the combined word.

Author: Removed the step "Combine all muscle geometries..." from this question. Also linked the step "Generate the fascia geometry" with its corresponding question.

Similarly to a skin simulation setup, the fascia can be simulated using [AdnSkin](maya/skin) deformer as well. In this case, it is recommended to use values of *Rest Length Multiplier* lower than 1. This deformer also requires one mesh (a single target, i.e. the muscle geometries combined) or multiple meshes (multiple targets, i.e. the muscle geometries separated) to drive the simulation and a mesh to apply the deformer to (fascia geometry).

## How can I generate the fascia geometry?

In order to obtain a plausible fascia geometry, we can use two techniques that require the muscle geometries and the skin geometry as inputs.

The first approach is to use the shrinkwrap deformer provided by Maya. This solution should produce good results for low poly geometries.

- Create a copy of the skin geometry that will become your fascia.
- Combine all muscle geometries into a single geometry.
- Select the combined muscles geometry, then select the copy of the skin geometry.
- Apply the shrinkwrap deformer and adjust its settings until the desired fascia geometry is achieved.
- Delete the history of the fascia geometry.

> Ths second approach is to use nCloth to apply a pressured simulation. This solution could work better for high-end fascia simulations. The procedure is the following:

Reviewer: it was my bad:

```suggestion
The second approach is to use nCloth to apply a pressured simulation. This solution could work better for high-end fascia simulations. The procedure is the following:
```

Author: Fixed.

## How can I generate the fascia geometry?

In order to obtain a plausible fascia geometry, we can use two techniques that require the muscle geometries and the skin geometry as inputs.

The first approach is to use the shrinkwrap deformer provided by Maya. This solution should produce good results for low poly geometries.

- Create a copy of the skin geometry that will become your fascia.
- Combine all muscle geometries into a single geometry. If an area is missing muscles, you can include some bone geometries along with the muscles to cover those areas.
- Select the combined muscles geometry, then select the copy of the skin geometry.
- Apply the shrinkwrap deformer and adjust its settings until the desired fascia geometry is achieved.
- Delete the history of the fascia geometry.

The second approach is to use nCloth to apply a pressured simulation. This solution could work better for high-end fascia simulations. The procedure is the following:

- Create a copy of the skin geometry that will become your fascia.
- Optionally, subdivide the fascia geometry if you need more resolution.

> - Combine all muscle geometries into a single geometry. If an area is missing muscles, you can include some bone geometries along with the muscles to cover those areas.

Author: Done.
